Python/pylifecycle.c

```diff
@@ -1292,6 +1292,7 @@ Py_InitializeEx(int install_sigs)
     config.install_signal_handlers = install_sigs;
 
     status = Py_InitializeFromConfig(&config);
+    PyConfig_Clear(&config);
```

Comment:
Aha, nice finding. Can you please write a separated PR for that, so it can be backported to other branches?

Reply:
For 3.11, I think it makes the most sense to backport the entire PR (since 3.11 also has the test changes this PR is built from), but I agree it makes sense to also backport the bug fix to 3.10. I'll add a second release note snippet to cover the functional tweak (since the 3.10 PR won't have the test suite change).

Reply:
Marking as resolved here based on adding the 3.11 backport label. I'll hold off on the 3.10 PR until after this one has been approved and merged.